frontend/views/components/modal/Modal.vue

```diff
@@ -58,13 +58,19 @@ export default ({
           if (this.content) this.replaceModal(toModal, omit(to.query, 'modal')) // if another modal is already open, replace it.
           else this.content = toModal
         }
-        const subcontent = to.query.subcontent ? to.query.subcontent.split('+').pop() : []
-        if (subcontent !== this.activeSubContent) {
-          // Try to find the new subcontent in the list of subcontent
-          const i = this.subcontent.indexOf(subcontent)
-          if (i !== -1) {
-            this.subcontent = this.subcontent.slice(0, i)
-          } else this.subcontent = subcontent
+
+        // Sometimes, the query string for modal has both 'modal' and 'subcontent' eg.) '/app/?modal=SignupModal&subcontent=Prompt'
+        if (to.query.subcontent) {
+          const subContentTo = to.query.subcontent.split('+').pop()
+          if (subContentTo !== this.activeSubContent) {
+            // Try to find the target subcontent in the list of subcontent
+            const i = this.subcontent.indexOf(subContentTo)
+            if (i !== -1) {
+              this.subcontent = this.subcontent.slice(0, i)
+            } else {
+              this.subcontent = [subContentTo]
+            }
+          }
```

Comment:
This is irrelevant to the issue itself but a fix for a bug I encountered while working on the issue.
The bug was that, when the app is reloaded and it has 'sign-up' modal and also 'login-error' prompt on top of it, `this.subcontent` becomes a weird string value (it was alphabet `t' in my case) when the entire relevant logic in `Modal.vue` is based on the assumption that it's an array of strings. So had to update this part with fixes like L:71 above.